### ISSUE TITLE
Type definition editor

### ISSUE THREAD

**Issue by oleksandr-dakal** [handsontable/handsontable#5046]
### Description
Type definition not correct for:
- Handsontable.cellTypes.registerCellType - missing
- Editor cellProperties must be `any `not an `object`. `object` type prevent access of custom properties.

### Your environment
* Handsontable version: 2.0.0
* Browser Name and version: Chrome 65.0.3325.181
* Operating System: Windows 10

**Comment by AMBudnik** [comment]
Hi @dakal-oleksandr 

Thanks for checking that.

The `any` case has been already raised and partially fixed with this PR https://github.com/handsontable/handsontable/pull/4689/files (it's waiting for review).

**Comment by dakal-oleksandr** [comment]
I found additionally that Handsontable.renderers.registerRenderer is missing

**Comment by aaronbeall** [comment]
Seems there's actually several things missing in the type definitions for top-level APIs:
```
Handsontable.cellTypes.getCellType()
Handsontable.cellTypes.registerCellType()
Handsontable.editors.getEditor()
Handsontable.editors.registerEditor()
Handsontable.validators.getValidator()
Handsontable.validators.registerValidator()
Handsontable.renderers.getRenderrer()
Handsontable.renderers.registerRenderer()
Handsontable.hooks.getRegistered() // Not missing but wrongly associated Events, appears on GridSettings
Handsontable.hooks... // a bunch of methods missing
```

**Comment by AMBudnik** [comment]
Closing this issue as well. As mentioned in the second post all the needed changes are made in https://github.com/handsontable/handsontable/pull/5767 (published under Handsontable v `7.0.0`).
